### DISCUSSION TITLE
HelloWorld on iOS fails to launch (11.0.0-preview4)

### DISCUSSION THREAD

**Discussion by Glacial** [AvaloniaUI/Avalonia#9768] [Q&A] [answered]
Hi all,
I'm checking in again to see if I can get anything running on iOS and once again I've hit a bit of a roadblock early on.

Using VS for Mac the HelloWorld template builds and run just fine in simulator as per the guide. Having set up a new provisioning profile edited the bundle identifier and modified the ios project for a real device (ios-arm64) I still run into trouble.
The app builds and deploys but crashes immediately on launch. In the analytics logs I can see an EXC_BAD_ACCESS error is raised but I don't understand enough at this point to gleam more. I've attached the ips for reference.

So what's going on this time? I'm hearing good things about Avalonia mobile but can't ever get off the ground. Please help.

[HelloWorld.iOS-2022-12-23-141642.ips.zip](https://github.com/AvaloniaUI/Avalonia/files/10295711/HelloWorld.iOS-2022-12-23-141642.ips.zip)

Update: I've since tried to go back to the simulator and find that this now fails in the same way. Can the provisioning profile cause this?

Update 2: The debugger has decided to throw and Exception I didn't change a thing:

> System.ExecutionEngineException: Attempting to JIT compile method '(wrapper managed-to-native) intptr Avalonia.OpenGL.GlBasicInfoInterface:wrapper_native_indirect_0x2821b83e0 (intptr&,int)' while running in aot-only mode. See https://docs.microsoft.com/xamarin/ios/internals/limitations for more information.
> 
>   at Avalonia.OpenGL.GlBasicInfoInterface.GetStringNative(Int32 v) in /_/src/Avalonia.OpenGL/DevGenerators/Generator.GetProcAddressInitializationGenerator/globalAvalonia.OpenGL.GlBasicInfoInterface.cs:13
>   at Avalonia.OpenGL.GlBasicInfoInterface.GetString(Int32 v) in /_/src/Avalonia.OpenGL/GlBasicInfoInterface.cs:27
>   at Avalonia.OpenGL.GlBasicInfoInterface.GetExtensions() in /_/src/Avalonia.OpenGL/GlBasicInfoInterface.cs:43
>   at Avalonia.OpenGL.GlInterface.GlContextInfo.Create(GlVersion version, Func`2 getProcAddress) in /_/src/Avalonia.OpenGL/GlInterface.cs:33
>   at Avalonia.OpenGL.GlInterface..ctor(GlVersion version, Func`2 getProcAddress) in /_/src/Avalonia.OpenGL/GlInterface.cs:48
>   at Avalonia.iOS.GlContext..ctor() in /_/src/iOS/Avalonia.iOS/EaglDisplay.cs:30
>   at Avalonia.iOS.EaglFeature..ctor() in /_/src/iOS/Avalonia.iOS/EaglDisplay.cs:17
>   at Avalonia.iOS.Platform.Register() in /_/src/iOS/Avalonia.iOS/Platform.cs:34
>   at at Avalonia.Controls.AppBuilderBase`1[[Avalonia.AppBuilder, Avalonia.Controls, Version=11.0.0.0, Culture=neutral, PublicKeyToken=c8d484a7012f9a8b]].Setup()
>   at at Avalonia.Controls.AppBuilderBase`1[[Avalonia.AppBuilder, Avalonia.Controls, Version=11.0.0.0, Culture=neutral, PublicKeyToken=c8d484a7012f9a8b]].SetupWithLifetime(IApplicationLifetime lifetime)
>   at at Avalonia.iOS.AvaloniaAppDelegate`1[[HelloWorld.App, HelloWorld, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].FinishedLaunching(UIApplication application, NSDictionary launchOptions)

**Comment by maxkatz6** [comment, marked as answer]
See https://github.com/dotnet/runtime/issues/80853

**Comment by danwalmsley** [comment]
https://github.com/AvaloniaUI/Solitaire/blob/master/Solitaire.iOS/Solitaire.iOS.csproj#L10

This is the workaround!
